tests/verify.rs

```diff
@@ -0,0 +1,41 @@
+const CRATE: &str = "test-flip-link-app";
+
+#[test]
+fn should_link_example_firmware() -> anyhow::Result<()> {
+    // Arrange
+    cargo::install_flip_link();
+
+    // Act
+    cargo::build_example_firmware()
+
+    // Assert
+}
+
+mod cargo {
+    use std::process::Command;
+
+    use assert_cmd::prelude::*;
+
+    use super::CRATE;
+
+    pub fn build_example_firmware() -> anyhow::Result<()> {
+        let mut firmware_dir = std::env::current_dir()?;
+        firmware_dir.push(CRATE);
+
+        Command::new("cargo")
+            .args(&["build", "--examples"])
+            .current_dir(firmware_dir)
+            .unwrap()
+            .assert()
+            .success();
+        Ok(())
+    }
+
+    pub fn install_flip_link() {
```

Comment:
hrmmm. this function seems smelly to me, but I can't quite put my finger on why. imo dependencies should generally be installed as part of the CI setup stage, but I see how that's not really feasible here (we need to make sure that the `flip-link` being used is the version that's in *this* repo). I'm wondering if it'd be possible to pass this repo's version of `flip-link` to `RUSTFLAGS` in the cargo build call in `build_example_firmware` to override `https://github.com/knurling-rs/flip-link/blob/main/test-flip-link-app/.cargo/config.toml#L4`

Reply:
That's a very good point! Also we should make it possible to run the test-suite against another version of flip-link. I guess I will remove it from here and create an `xtask` wrapper containing the installation (in a next PR).